test/eth/TransactionManager.spec.js

```diff
@@ -227,13 +229,15 @@ describe('lifecycle hooks', () => {
     // after calling confirm, Tx state will become 'finalized' and be deleted from list.
     await Promise.all([txMgr.confirm(open), mineBlocks(service)]);
 
-    expect(txMgr._tracker._transactions).not.toHaveProperty(openId);
+    txMgr._tracker.clearExpiredTransactions();
```

Comment:
@b-pmcg @sirromdev I just wanted to point out this one because it caused me some prolonged puzzlement. 

So this test and the "clear Tx when state is error" test after it were failing.  Patrick had added a [workaround in TransactionManager](https://github.com/makerdao/dai.js/pull/63/files#diff-7adfe82c3110f0ada2d623169f37c78eR91) to make them pass, but I removed it, because I couldn't explain to myself why it was needed. (General principle here: be wary of changing code to make a test pass if you don't understand why it started failing.)

But then as I looked at the code more closely, I couldn't figure out why it had been working in the first place, since the only place we call `clearExpiredTransactions` is in `TransactionManager.tracker.store`...

Finally it clicked -- I'd changed the implementation of `mineBlocks` to call an `evm_mine` command on Ganache directly. Seemed more elegant, with no side effects. (Nice find, btw, Patrick.) But that meant `mineBlocks` was no longer sending new transactions through the library, hence no longer calling `TransactionManager.tracker.store`. The tests had been relying upon that side effect.

Reply:
Yes, I believe I came across this a week or two ago. I cannot recall the finer details 100% but I did make [this](https://github.com/makerdao/dai.js/pull/63/commits/d33219ff62c29fd2777cd77e5b22457096ddb1f8#diff-7adfe82c3110f0ada2d623169f37c78eR94) change to ensure that transactions are cleaned up on a more regular basis

Reply:
For "evm_mine", it would possibly be a good idea to have a convenience method to use this to iterate blocks without having to issue actual transactions. It is definitely preferable in some situations to issue transactions to propagate blocks as is done in `mineBlocks` but I have used it [here](https://github.com/makerdao/dai.js/pull/63/commits/bdedeea5a6f4b09a3754fc9c36986eae1d136e8e#diff-db1ade81f846b7ee45b14e3eaf5f5299R478) to disassociate ganache behaviour with functions of transaction system.

Reply:
Yes, that’s the change I reverted 😅 In ordinary usage of the library, having `clearExpiredTransactions` be called from `store` is fine. There’s no need to add it in `confirm` as well.

I don’t think there’s any need to preserve the old behavior of `mineBlocks` either. We absolutely should not be depending upon an unintentional side effect of it on the library when all we want it to do is make more blocks appear.